### ISSUE TITLE
[BUG] The program doesn't start. missing icon.

### ISSUE THREAD

**Issue by neominky** [vrchat-community/creator-companion#444]
### Describe the bug

windows event viewer logged

Application: CreatorCompanion.exe
CoreCLR Version: 6.0.2724.6912
.NET Version: 6.0.27
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ArgumentException: Icon file:  does not exist.
   at PhotinoNET.PhotinoWindow.set_IconFile(String value)
   at VCCApp.Program.Main(String[] args)

The same applies to reinstalling.

### To Reproduce

just click CreatorCompanion icon on start menu.
or
just duble click C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\CreatorCompanion.exe on explorer

### Expected behavior

missing icon in package

### Related Product

VCC (VRChat Creator Companion)

### Product Version

2.2.5

### OS

Windows 11

### Additional context

file lists in installed folder
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\CreatorCompanion.exe
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\list.txt
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\nfd.dll
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\nfd.exp
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\nfd.lib
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Photino.Native.dll
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Tools
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\unins000.dat
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\unins000.exe
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebView2Loader.dll
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\Assets
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\package.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\Packages
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\README.md
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\Assets\.gitkeep
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\Packages\.gitignore
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\Packages\manifest.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\Packages\vpm-manifest.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\AudioManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\ClusterInputManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\DynamicsManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\EditorBuildSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\EditorSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\GraphicsSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\InputManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\MemorySettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\NavMeshAreas.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\PackageManagerSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\Physics2DSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\PresetManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\ProjectSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\ProjectVersion.txt
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\QualitySettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\TagManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\TimeManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\UnityConnectSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\VersionControlSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\VFXManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar\ProjectSettings\XRSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\Assets
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\package.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\Packages
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\README.md
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\Assets\.gitkeep
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\Packages\.gitignore
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\Packages\manifest.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\Packages\vpm-manifest.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\AudioManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\ClusterInputManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\DynamicsManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\EditorBuildSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\EditorSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\GraphicsSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\InputManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\NavMeshAreas.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\PackageManagerSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\Physics2DSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\PresetManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\ProjectSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\ProjectVersion.txt
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\QualitySettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\TagManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\TimeManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\UnityConnectSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\VFXManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Avatar 2019\ProjectSettings\XRSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\Assets
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\package.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\Packages
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\README.md
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\Assets\.gitkeep
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\Packages\.gitignore
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\Packages\manifest.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\Packages\vpm-manifest.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\AudioManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\ClusterInputManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\DynamicsManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\EditorBuildSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\EditorSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\GraphicsSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\InputManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\MemorySettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\NavMeshAreas.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\PackageManagerSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\Physics2DSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\PresetManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\ProjectSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\ProjectVersion.txt
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\QualitySettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\TagManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\TimeManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\UnityConnectSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\VersionControlSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\VFXManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\Base\ProjectSettings\XRSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\Assets
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\package.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\Packages
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\README.md
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\Assets\.gitkeep
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\Packages\.gitignore
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\Packages\manifest.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\Packages\vpm-manifest.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\AudioManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\ClusterInputManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\DynamicsManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\EditorBuildSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\EditorSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\GraphicsSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\InputManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\MemorySettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\NavMeshAreas.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\PackageManagerSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\Physics2DSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\PresetManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\ProjectSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\ProjectVersion.txt
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\QualitySettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\TagManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\TimeManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\UnityConnectSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\VersionControlSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\VFXManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World\ProjectSettings\XRSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\Assets
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\package.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\Packages
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\README.md
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\Assets\.gitkeep
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\Packages\.gitignore
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\Packages\manifest.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\Packages\vpm-manifest.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\AudioManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\ClusterInputManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\DynamicsManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\EditorBuildSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\EditorSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\GraphicsSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\InputManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\NavMeshAreas.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\PackageManagerSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\Physics2DSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\PresetManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\ProjectSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\ProjectVersion.txt
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\QualitySettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\TagManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\TimeManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\UnityConnectSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\VFXManager.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Templates\World 2019\ProjectSettings\XRSettings.asset
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Tools\VRC Quick Launcher
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\Tools\VRC Quick Launcher\VRC Quick Launcher.exe
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\assets
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\buildInfo.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\crash.html
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\favicon.ico
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\icon.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\index.html
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\learn
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\learn.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\logs.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\manifest.json
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\projects.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\robots.txt
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\settings.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\template-covers
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\tools.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\vrcql-icon.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\vrcql.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\assets\index-48df1483.js
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\assets\index-50c696ec.css
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\assets\vendor.LICENSE.txt
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\learn\learn-avatars.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\learn\learn-discord.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\learn\learn-engineUpgrade.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\learn\learn-forums.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\learn\learn-graph.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\learn\learn-vcc.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\learn\learn-worlds.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\template-covers\template-avatar-2022.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\template-covers\template-avatar.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\template-covers\template-custom.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\template-covers\template-world-2022.png
C:\Users\username\AppData\Local\Programs\VRChat Creator Companion\WebApp\Dist\template-covers\template-world-udon.png

**Comment by gwanryo** [comment]
```
Application: CreatorCompanion.exe
CoreCLR Version: 6.0.2724.6912
.NET Version: 6.0.27
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ArgumentException: Icon file:  does not exist.
   at PhotinoNET.PhotinoWindow.set_IconFile(String value)
   at VCCApp.Program.Main(String[] args)
```
I'm encountering the same issue on Windows 10 Version 22H2 (OS Build 19045.4291)

**Comment by orels1** [comment]
@gwanryo could you try out the latest beta build? It [can be downloaded using this link](https://github.com/vrchat-community/creator-companion/releases/download/2.3.0-beta.2/web_vcc_2.3.0-beta.2_Release_Portable.zip)

**Comment by gwanryo** [comment]
@orels1 I've reinstalled VCC 2.2.5 (not beta) a couple of times, and suddenly it's working fine. I'm not exactly sure what caused the issue. I also downloaded the link you provided and ran it, and everything seems to be working smoothly now.

**Comment by orels1** [comment]
Good to know it works for you again.
It reads the icon from here `WebApp\Dist\favicon.ico` so if you ever encounter this same issue, I'd ask to check if that icon is in place or not 🙏

**Comment by repryty** [comment]
The same error occurs even though the favicon.ico file is properly located in that location. I tried running the beta above and it's the same.

**Comment by orels1** [comment]
@repryty same logs too? That is concerning.
Could I ask you to zip-up the folder with the beta install that fails on launch for you and send it to orels@vrchat.com?

**Comment by JustArion** [comment]
@orels1 From looking at the code, there is 1 flaw that is apparent. Null checks aren't being done.

The process silently crashes due to the following fields being null when the app crashes:
```cs
PhotinoWindow photinoWindow = new PhotinoWindow()
{
    IconFile = Program.IconPath,
    Title = Program.WindowTitle,
    UseOsDefaultSize = true,
    Resizable = true
};
```
That being `Program.IconPath`.

The `RunVCC` method has 45 lines of code before `Program.IconPath` is set. That's 45 lines where a point of failure can occur where it will silently crash.

My proposal would be to have the IconPath either be set earlier (like near the start of the method) or to use the `Lazy<T>` class to set the value.
eg.
```cs
private static string IconPath = new Lazy<string>(()=> 
{
    Program.IconPath = Path.Combine("WebApp", "Dist");
    Program.IconPath = !RuntimeInformation.IsOSPlatform(OSPlatform.Windows) 
        ? Path.Combine(Program.IconPath, "icon.png")
        : Path.Combine(Program.IconPath, "favicon.ico");
});
```

Logging is also not being set up before the error. Critical infrastructure should be called first.

The actual error hidden behind all this (At least in my case) is `An attempt was made to access a socket in a way forbidden by its access permissions.`
from this piece of code:
```cs
Program.WebApiPort = VCCApp.Extensions.Helpers.GetAvailableTcpPort(Program.WebApiPort);
Program.ApiServerPort = VCCApp.Extensions.Helpers.GetAvailableTcpPort(Program.ApiServerPort);
```

In my case, this is likely related to Mullvad's VPN KillSwitch. Turning off the VPN fixed CreatorCompanion.


As a take-away, for the VRC Team please keep Nullable Reference Types enabled 🙃

**Comment by orels1** [comment]
Hey, while I appreciate the information, I would also ask to not discuss decompiled code in a public setting, that limits how I can respond 😅 

In terms of the rest, yes that seems like the core issue is the port acquisition as there isn't really much else that could fail there, and I wasn't really aware could fail.
And the fallback to the "things crashed" window didn't work because of the Icon, as you noted.

We'll fix it for the next release, thank you

**Comment by orels1** [comment]
This has been fixed internally and will ship with 2.3.0
